core/modules/css/KvInterfaceImplMem.cc

```diff
@@ -79,14 +82,18 @@ namespace css {
   *    mv /tmp/testMap.kvmap <destination>
   */
 KvInterfaceImplMem::KvInterfaceImplMem(std::istream& mapStream, bool readOnly)
-    : _readOnly(readOnly) {
+    : _readOnly(false) {
     _init(mapStream);
+    // read-only should only be set after _init
+    _readOnly = readOnly;
 }
 
 KvInterfaceImplMem::KvInterfaceImplMem(string const& filename, bool readOnly)
-    : _readOnly(readOnly) {
+    : _readOnly(false) {
```

Comment:
I think it would be better to set _readOnly{false}; in the header file. It might be possible to call the other constructor directly, but I'm not sure if : KvInterfaceImplMem(std::ifstream f(filename), readOnly) will work.

Reply:
I'll move _readOnly init to header; as for delegating constructor - that would not work, you cannot bind temporary to non-const reference.